pandas/core/series.py

```diff
@@ -1348,16 +1352,34 @@ def multi(values, qs, **kwargs):
                 values = [_quantile(values, x * 100, **kwargs) for x in qs]
                 # let empty result to be Float64Index
                 qs = Float64Index(qs)
-                return self._constructor(values, index=qs, name=self.name)
+                return qs, np.array(values)
             else:
-                return _quantile(values, qs * 100, **kwargs)
+                return None, _quantile(values, qs * 100, **kwargs)
 
         kwargs = dict()
         if not _np_version_under1p9:
             kwargs.update({'interpolation': interpolation})
 
-        return self._maybe_box(lambda values: multi(values, q, **kwargs),
-                               dropna=True)
+        values = self.dropna()._values
+
+        if len(values) == 0:
+            return self._data._block._na_value
+
+        if (com.is_datetime_or_timedelta_dtype(self) or
+           com.is_datetime64tz_dtype(self)):
+
+            values = values.view('i8')
+            index, result = multi(values, q, **kwargs)
+            result = self._data._block._try_coerce_result(result,
```

Comment:
as an aside. It will just be simpler to move the quantiling into an core/internal method directly to avoid this kind of messiness. (e.g. you can simply do: ``self._data.quantile(q)`` (and it will automatically block process and you can handle the coercion internally). 

This might be a bit of effort though.